todo/src/App.js

```diff
@@ -0,0 +1,44 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { addTodo } from './actions';
+import TodoList from './components/TodoList';
+
+class App extends Component {
+  state = {
+    todo: ''
+  }
+
+  changeHandler = e => {
+    this.setState({ [e.target.name]: e.target.value });
+  }
+
+  submitHandler = todo => {
+    this.props.addTodo(todo);
```

Comment:
```suggestion
   e.preventDefault()
    this.props.addTodo(this.state.todo);
```
So this is a common misconception. You're actually invoking the addTodo action creator inside of submithandler, so rather than setting a variable inside addTodo you actually want to pass the value of something real into this here.